src/app/projects/project-details/project-details-build/project-details-build.component.ts

```diff
@@ -86,22 +86,30 @@ export class ProjectDetailsBuildComponent {
     }
   }
 
+  numberWithCommas(num: number) {
+    return num.toString().replace(/\B(?=(\d{3})+(?!\d))/g, ',');
+  }
```

Comment:
[`Intl.NumberFormat().format`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Intl/NumberFormat) is available [in most browsers](https://caniuse.com/mdn-javascript_builtins_intl_numberformat_format)

```js
const number = 123456;
console.log(new Intl.NumberFormat('en-US').format(number));
// > 123,456
```

Suggest using that instead, as it's more readable than regex magic:

```suggestion
  numberWithCommas(num: number) {
    return new Intl.NumberFormat('en-US').format(num);
  }
```

Or perhaps to micro-optimize then you save the format as a class member so it doesn't have to lookup the format each time.

```diff
@@18-27@@
 export class ProjectDetailsBuildComponent {
   @Input() project: WharfProject;
   rowsCount = 10;
   buildsTotalCount = 0;
   excludedElements = ['environments', 'inputs'];
   selectedAction: string;
   actionsModalVisible = false;
   isActionsFormVisible = false;
   destroyed$ = new Subject<void>();
+
+  private numberFormat = new Intl.NumberFormat('en-US');

@@88-92@@

   numberWithCommas(num: number) {
-    return num.toString().replace(/\B(?=(\d{3})+(?!\d))/g, ',');
+    return this.numberFormat.format(num);
   }
```

Reply:
Oooh, neat.